mediacapture-streams/MediaDevices-getUserMedia.https.html

```diff
@@ -37,60 +35,57 @@ <h1 class="instructions">Description</h1>
   //   list.groupId
   }, "mediaDevices.getUserMedia() is present on navigator");
 
-promise_test(t => {
-  return navigator.mediaDevices.enumerateDevices()
-    .then(t.step_func(async devices => {
-      for (var i in devices) {
-        await navigator.mediaDevices.getUserMedia(
-          {video: {groupId: {exact: devices[i].groupId}}})
-            .then(
-              t.step_func(stream => {
-                var found_device = devices.find(element => {
-                  return element.deviceId ==
-                           stream.getTracks()[0].getSettings().deviceId;
-                });
-                assert_true(undefined !== found_device);
-                assert_equals(found_device.kind, "videoinput");
-                assert_equals(found_device.groupId, devices[i].groupId);
-              }),
-              t.step_func(error => {
-                assert_equals(error.name, "OverconstrainedError");
-                assert_equals(error.constraint, "groupId");
-                var found_device = devices.find(element => {
-                  return element.kind == "videoinput" &&
-                        element.groupId == devices[i].groupId});
-                assert_true(undefined === found_device);
-              }));
-      }
-    }));
+promise_test(async t => {
+  const devices = await navigator.mediaDevices.enumerateDevices();
+  const tracks = [];
+  t.add_cleanup(() => tracks.forEach(track => track.stop()));
+  for (const i in devices) {
+    try {
+      const stream = await navigator.mediaDevices.getUserMedia(
+        {video: {groupId: {exact: devices[i].groupId}}}
+      );
+      tracks.push(...stream.getTracks());
+      const found_device = devices.find(element => {
+        return element.deviceId == stream.getTracks()[0].getSettings().deviceId;
+      });
+      assert_true(undefined !== found_device);
+      assert_equals(found_device.kind, "videoinput");
+      assert_equals(found_device.groupId, devices[i].groupId);
+    } catch (e) {
+      assert_equals(e.name, "OverconstrainedError");
```

Comment:
Are you supposed to hit this branch, or not?
If you sometimes hit it, sometimes not, please make a comment to that effect.
If you expect to hit it every time, please find a way to fail the test if you don't.

Reply:
This test traverses all available devices and part of them maybe overconstrained if it does not a video device, that's why it catches the error and make it as expected result. And it's the original logic, I just rewrote it with `async/await`.

Reply:
I will fix it in the next PR.